File: 2023/UICPC DIV1/UICPC Round #04/solutions/A-PlantingTrees/trees_lukas.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#Solution by lukasP (Lukáš Poláček)
from sys import stdin

n = int(stdin.readline())
a = map(int, stdin.readline().split())
a.sort()
a.reverse()

print max(i + a[i] + 2 for i in range(n))

File: 2023/UICPC DIV1/UICPC Round #04/solutions/A-PlantingTrees/trees_lukas.cs
//Solution by lukasP (Luká¿ Polá¿ek)
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;

class Problem
{
    public static void Main ()
    {
        var bs = new BufferedStream(Console.OpenStandardInput());
        StreamReader reader = new StreamReader(bs);

        int n = int.Parse(reader.ReadLine());
        List<int> a = reader.ReadLine().Split().Select(x => int.Parse(x))
            .OrderByDescending(x => x).ToList();

        Console.WriteLine(Enumerable.Range(0, n).Max(i => i + a[i] + 2));
    }
}
